include/mbedtls/ccm.h

```diff
@@ -85,6 +85,8 @@ void mbedtls_ccm_init( mbedtls_ccm_context *ctx );
  *
  * \return          \c 0 on success.
  * \return          A CCM or cipher-specific error code on failure.
+ * \return          #MBEDTLS_ERR_CCM_FEATURE_UNAVAILABLE if the
+ *                  alternative implementation doesn't support the key size or cipher mode.
```

Comment:
For what reason? The features that an alternative implementation cannot support are the key size and the cipher type.
Both are set in `mbedtls_ccm_setkey()`.
Perhaps the length is too big for the HW to handle.

Reply:
There could be a glitch in the power supply, or the hardware is in use doing another operation and can't begin a new operation, or hardware has a bug and the driver for the software is preventing use of the hardware for this function as a "solution" to prevent people from using buggy hardware.

Reply:
we have [`MBEDTLS_ERR_AES_HW_ACCEL_FAILED`](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/aes.h#L64) for any HW failure, which is a real failure. The FEATURE_UNAVAILBE error is a valid error, that tells the user that this feature is not suipported by the HW.